openedx_events/learning/data.py

```diff
@@ -469,3 +469,53 @@ class CourseNotificationData:
     content_url = attr.ib(type=str)
     content_context = attr.ib(type=dict, factory=dict)
     audience_filters = attr.ib(type=dict, factory=dict)
+
+
+@attr.s(frozen=True)
+class UserCourseData:
+    """
+    Attributes defined for Open edX user course data object.
+
+    Arguments:
+        user (UserData): user associated with the Course Enrollment.
+        course (CourseData): course where the user is enrolled in.
+    """
+
+    user = attr.ib(type=UserData)
+    course = attr.ib(type=CourseData)
```

Comment:
What's the reason for adding this Data class? It seems redundant when we can get the same effect in other data classes by including `UseData` and `CourseData` data objects in the grade's event data dict. 

In the newly created signals, the data object could just be...
```
event_type="blahblahblah",
data = {
    "user": UserData,
    "course": CourseData"
}
```

To me, this seems like we're adding a level of nesting and I can't see a clear benefit at face value.

Reply:
Instead of this just being a combination of `UserData` and `CourseData`, should we instead possibly make this a `GradeData` data class?

Since this is being used in the new `COURSE_GRADE_NOW_PASSED` and `COURSE_GRADE_NOW_FAILED` events, it could be useful to the folks who may be consuming this event in the future to have the actual grade data available as part of the event. 

Hopefully we could include some items like `percent_grade`, `letter_grade` and other model fields from the `PersistentCourseGrade` model of the LMS? 

Seems like we're missing an opportunity here to include that data for future event bus consumers.

Reply:
Even going a bit further, it looks like we already have a data object for `PersistentCourseGradeData`, which I think would be a useful thing to pass along as the event data for the above PASSED/FAILED events.